src/components/ModalDetails/ModalDetails.js

```diff
@@ -376,20 +381,13 @@ const ModalDetails = ({
             defaultValue={datePickerValue}
             disabled={isEditingDisabled}
             maxDate={
-               // Allow future dates up to 90 days for specific components
-              ["Default", "Formula", "Load", "Location Attribute", "Method",
-                "Qualification", "Rectangular Duct WAF", "Span", "System",
-                "Unit Information", "Unit Fuel", "Unit Control", "Unit Capacity"].includes(title)
-                ? (() => {
-                    const futureDate = new Date(d.getTime() + (90 * 24 * 60 * 60 * 1000));
-                    return `${futureDate.getFullYear()}-${futureDate.getMonth() + 1}-${futureDate.getDate()}`;
-                  })()
-
-              : title !== "Protocol Gas"
-                ? `${d.getFullYear()}-${d.getMonth() + 1}-${d.getDate()}`
-                : null
-            }
-          />
+              FUTURE_DATE_ALLOWED_COMPONENTS.includes(title)
```

Comment:
The new prop would be used here:
```suggestion
              allowFutureDates
```

Reply:
updated. see comments below.